cmd/thanos/sidecar.go

```diff
@@ -81,16 +80,22 @@ func runSidecar(
 	gcsBucket string,
 ) (okgroup.Group, error) {
 
-	var extLabelsMu sync.RWMutex
-	var externalLabels labels.Labels
+	var g okgroup.Group
+	externalLabels := &extLabelSet{promURL: promURL}
 
-	getExternalLabels := func() labels.Labels {
-		extLabelsMu.Lock()
-		defer extLabelsMu.Unlock()
-		return externalLabels
+	// Get external labels before anything else.
+	{
+		ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+		defer cancel()
```

Comment:
I wonder whether we should have a timeout here. This makes us actually depend on startup order. Depending on environment, this may be longer than 15 seconds.

Maybe we should just retry indefinitely and print a warning each time we did not succeed?

When we eventually implement readiness endpoints that would just return a non-200 until it succeeded.

Reply:
makes sense